### PR TITLE
DEV2-2085 logs tooling

### PR DESCRIPTION
**Pull request by dimacodota**
new `tabnine.logEngine` setting introduced, by enabling  it the binary  stderr will be outputted to the extension log

Screen Shot 2023-07-19 at 23 45 40

### CODE REVIEW

**Review by assapir — APPROVED** [review]

File: src/utils/logger.ts [hunk 1]
```diff
@@ -11,12 +11,23 @@ enum LogLevel {
...
   constructor() {
     this.outputChannel = vscode.window.createOutputChannel("Tabnine");
+    this.showLogsDisposable = vscode.commands.registerCommand(
+      "tabnine.logs",
+      () => this.show()
+    );
+  }
+
+  init(context: vscode.ExtensionContext) {
+    context.subscriptions.push(this);
```
Comment (assapir) [inline src/utils/logger.ts:25]: why is that needed?

File: src/utils/logger.ts [hunk 2]
```diff
@@ -40,6 +51,10 @@ export const Logger = new (class Logger implements vscode.Disposable {
     this.log(LogLevel.DEBUG, message, optionalParams);
   }
 
+  process(message: string): void {
+    this.outputChannel.appendLine(`${message}\n`);
```
Comment (assapir) [inline src/utils/logger.ts:55]: I think it's better to have a prefix or something. this might cause a lot of noise and we should be able to tell if that a binary log or the extension log

File: src/binary/runProcess.ts [hunk 3]
```diff
@@ -29,6 +29,10 @@ export function runProcess(
   );
   const proc = args ? spawn(command, args, options) : spawn(command, options);
 
+  proc.stderr?.on("data", (data: string) => {
+    Logger.process(data.toString());
```
Comment (assapir) [inline src/binary/runProcess.ts:33]: data is already string, why the need for `toString()`?

**Review by assapir — APPROVED** [review]
Great idea! I wonder if we can have a "zip logs and send to support" somehow.

File: src/utils/logger.ts (hunk 1 above)
Comment (assapir) [reply src/utils/logger.ts:25]: ok, I read about it. cool.

**Comment by dimacodota** [comment]
> Great idea! I wonder if we can have a "zip logs and send to support" somehow.
yep, i taught to add a "profile extension"  command that will collect the logs
